eclair-core/src/main/scala/fr/acinq/eclair/payment/PaymentRequest.scala

```diff
@@ -203,37 +203,37 @@ object PaymentRequest {
     * Hidden hop
     *
     * @param pubkey          node id
-    * @param channelId       channel id
+    * @param shortChannelId  channel id
     * @param fee             node fee
     * @param cltvExpiryDelta node cltv expiry delta
     */
-  case class HiddenHop(pubkey: PublicKey, channelId: BinaryData, fee: Long, cltvExpiryDelta: Int) {
-    def pack: Seq[Int5] = pubkey.toBin ++ channelId ++ Protocol.writeUInt64(fee, ByteOrder.BIG_ENDIAN) ++
-      Protocol.writeUInt16(cltvExpiryDelta, ByteOrder.BIG_ENDIAN)
+  case class ExtraHop(pubkey: PublicKey, shortChannelId: Long, fee: Long, cltvExpiryDelta: Int) {
```

Comment:
Is there a reason to represent a `shortChannelId` as `BinaryData` here (and named `channelId`)?
I'm trying to further work on processing payment requests with extra routing and it's much more convenient to have it represented as Long.

Reply:
Yes, using a Long here would be simpler